### DISCUSSION TITLE
External Grids with Negative Flow

### DISCUSSION THREAD

**Discussion by rohrerph** [e2nIEE/pandapipes#631] [Q&A]
Hi everyone,

I have created a gas network that connects to two external grids. When I run the network simulation, everything functions correctly, with both external grids supplying mass flow into my network. However, I've encountered an issue when I scale down the load of the sinks or switch off some sinks. The total required mass flow decreases, resulting in one external grid supplying gas into the system (positive flow) while the other external grid experiences a negative flow, indicating that gas is exiting the system. This effectively means the network is transporting gas through the system without consuming it and then releasing it back out, which is not the desired behavior.

Both external grids have their pressures set equally. Has anyone experienced a similar issue or knows how to resolve it so that gas can only enter the network through the external grids and cannot flow back out?

Thanks for your help!

**Comment by jkisse** [comment]
Interesting question. I think it is caused by the hydraulic situation in the grid and probably the only feasible solution due to the two fixed pressures. Are there any sources in the network? 
Even if the 2nd external grid could be modified in a way that it doesn't serve as a mass flow slack, only fixes the pressure at the junction, it would probably just lead to a non-feasible hydraulic situation in the network. 

The only solution I can think of right now is to increase the pressure of the 2nd external grid to a level at which no mass flow is "drawn" out of the network there. Decreasing the 1st external grid's pressure could also work, if the pressures do not become negative in that case.

Maybe someone else can support with deeper insights, hopefully, or experience from other tools.

**Comment by dlohmeier** [comment]
Hi @rohrerph ,
I have not encountered such a situation by now. Could you provide an example for this kind of behavior? It does sound odd, especially if you do not define any sources and the pressure level of the two external grids is exactly equal.
